### DISCUSSION TITLE
Disable exponential backoff in curl when using the --retry option

### DISCUSSION THREAD

**Discussion by AntonioL** [curl/curl#10809] [Q&A] [answered]
I am finding that the following combination of parameters in curl does not eliminate the exponential backoff strategy when it has to retry requests.

```bash
curl \
     --retry 5 --retry-all-errors \
     --retry-delay 0 \
     --max-time 1 \
     --connect-timeout 1 \
     --trace-time \
     --trace-ascii /tmp/report.txt \
    -O -L https://httpbin.org/delay/10
```
Per the cURL documentation this should do it: -`-retry-delay <seconds> Wait time between retries`

Apparently setting it to 0 does not disable exponential backoff. If I set it to 1 it will attempt each request 1 second after but that is not what I want. Anything I am missing in my combination of parameters?

I am using cURL version 7.88.1

**Comment by jay** [comment]
I'm not sure what you are reporting. [`--retry-delay <seconds>`](https://curl.se/docs/manpage.html#--retry-delay) if 0 then the default time is used as noted in the doc. Currently it defaults to 1 second (with exponential backoff). It's no different than not using it at all.

https://github.com/curl/curl/blob/b16d1fa8ee567b52c09a0f89940b07d8491b881d/src/tool_main.h#L30-L31

**Reply by AntonioL** [reply]
I need --retry-delay to be non existent (0 seconds). Is that possible?

**Reply by jay** [reply, marked as answer]
Sorry no. You could handle retries in shell script or use a custom built curl with RETRY_SLEEP_DEFAULT 0.

Also note:

_"Since curl 7.66.0, curl will comply with the Retry-After: response header if one was present to know when to issue the next retry."_ Though it will not exceed the user set maximum allowed time.
